documentation/guides/02_dependencies.md

```diff
@@ -53,38 +46,121 @@ For the first imported dependency, a new `dependencies` field will be added to t
   "dependencies": [
     {
-      "name": "local_dependency.aleo",
+      "name": "my_library.aleo",
       "location": "local",
       "network": null,
-      "path": "./path"
+      "path": "./path/to/my_library"
     }
   ]
-} 
+}
+```
+
+Local dependencies are compiled from source whenever you build. They never require network access.
+
+### Removing Dependencies
+```bash
+leo remove credits.aleo
+```
+
+## Using Dependencies in Code
+
+In your `main.leo`, import dependencies before the program declaration:
+```leo
+import credits.aleo;
+import my_library.aleo;
+
+program my_program.aleo {
+    // ...
+}
+```
+
+## How Dependency Resolution Works
+
+When you run a Leo command, dependencies are resolved as follows:
+
+1. **Read `program.json`** to find declared dependencies
+2. **For each dependency:**
+   - **Local**: Read the Leo source from the specified path and compile it
+   - **Network**: Fetch the bytecode from the Aleo network (or cache)
+3. **Resolve transitive dependencies** - if your dependency imports other programs, those are fetched too
+4. **Topologically sort** all programs so dependencies are processed before dependents
+
+### Caching Behavior
+
+Network dependencies are cached locally at `~/.aleo/registry/{network}/{program_name}/{edition}/` to avoid repeated downloads.
+
+Different commands handle caching differently:
+
+| Command | Cache Behavior |
+|---------|---------------|
+| `leo build` | Uses cache |
+| `leo run` | Uses cache |
+| `leo execute` | Always fetches fresh |
+| `leo deploy` | Always fetches fresh |
+| `leo upgrade` | Always fetches fresh |
+| `leo synthesize` | Always fetches fresh |
+
+Commands that generate proofs (`execute`, `deploy`, `upgrade`, `synthesize`) always fetch fresh bytecode because proofs include commitments to the exact bytecode of dependencies. Using stale cached bytecode would produce invalid proofs if a dependency has been upgraded on-chain.
+
+To force a fresh fetch during `build` or `run`:
+```bash
+leo build --no-cache
 ```
 
-## Recursive Deployment
-When deploying a program that uses local dependencies, use the following command:
+## Editions (Program Versions)
+
+An **edition** is the version number of a deployed program on the Aleo network:
+- Edition 0: Initial deployment
+- Edition 1: First upgrade
+- Edition 2: Second upgrade
+- ...and so on
+
+By default, Leo fetches the **latest** edition of a network dependency. To pin to a specific edition:
+```bash
+leo add some_program.aleo --edition 3
+```
+
+This records the pinned edition in the manifest:
+```json
+{
+  "name": "some_program.aleo",
+  "location": "network",
+  "network": "testnet",
+  "path": null,
+  "edition": 3
+}
+```
+
+**When to pin editions:**
+- When you need reproducible builds
+- When a dependency upgrade would break your program
+- When you want to avoid unexpected behavior changes
+
+**Note:** Local dependencies don't have editions - they're always compiled from your current source code.
```

Comment:
What happens when the local dependency also lives in the network? 

In any case, it must have some edition right, either 0 if this is the first deployment or perhaps higher if it's already present?

Not sure if the edition is actually materially used for anything.